### ISSUE TITLE
portal.processHtml does not convert FileURIs

### ISSUE THREAD

**Issue by ComLock** [enonic/xp#5796]
I think processHtml should convert FileURIs according to this article:
https://blogs.msdn.microsoft.com/ie/2006/12/06/file-uris-in-windows/

This is failing at one of our customers which use links from their intranet site onto windows file shares.

**Comment by sigdestad** [comment]
Can you provide an example of a "fileURI"? What do you want to achieve?

**Comment by alansemenov** [comment]
@ComLock what do we do with this one?

**Comment by ComLock** [comment]
@sigdestad 
If you paste in either of these into chrome url the file will open fine in the browser:
file://\\mapping\path\filename.html
file://C:\Path\filename.html

If you paste them into the url field for a link in the htmlarea editor and run processHtml on it you will end up with illegal non working urls, so clicking the link will not open in the browser.

Untested:

You end up with these illegal urls:
file://\\\\mapping\\path\\filename.html
file://C:\\Path\\filename.html

The correct ones should be:
file://mapping/path/filename.html
file:///C:/Path/filename.html

**Comment by ComLock** [comment]
Yes they are actually linking to html files stored on network shares.

**Comment by mortyeriksen** [comment]
We need to check if fixing this cause any security issues.

**Comment by rymsha** [comment]
In modern browsers file URI don't wrk from web pages - due to security reasons.

It is possible to "enable it back" via specific browser settings - but this feature is very niche and insecure to fully support it.
